x/shield/keeper/proposal.go

```diff
@@ -45,7 +52,7 @@ func (k Keeper) ClaimLock(ctx sdk.Context, proposalID uint64, poolID uint64, los
 	if purchase.ExpirationTime.Before(votingEndTime) {
 		purchase.ExpirationTime = votingEndTime
 	}
-	k.SetPurchase(ctx, purchase)
+	k.AddPurchase(ctx, poolID, purchaser, purchase)
```

Comment:
Should we remove the obsolete purchase before adding the updated one? Or maybe update `purchaseList` directly like you did in `RestoreShield`.

Reply:
Yes definitely. This might have been the cause of the error I had earlier